<launch>

  <arg name="bagfiles" default="/home/xesc/BAGFILES_BAG/emisari_27092017/images170927_1234_2017-09-27-10-54-01_36.bag
    /home/xesc/BAGFILES_BAG/emisari_27092017/images170927_1234_2017-09-27-10-54-33_37.bag
    /home/xesc/BAGFILES_BAG/emisari_27092017/images170927_1234_2017-09-27-10-55-05_38.bag
    /home/xesc/BAGFILES_BAG/emisari_27092017/images170927_1234_2017-09-27-10-55-37_39.bag
    /home/xesc/BAGFILES_BAG/emisari_27092017/images170927_1234_2017-09-27-10-56-09_40.bag
    /home/xesc/BAGFILES_BAG/emisari_27092017/images170927_1234_2017-09-27-10-56-41_41.bag
    /home/xesc/BAGFILES_BAG/emisari_27092017/images170927_1234_2017-09-27-10-57-13_42.bag
    /home/xesc/BAGFILES_BAG/emisari_27092017/images170927_1234_2017-09-27-10-57-45_43.bag
    /home/xesc/BAGFILES_BAG/emisari_27092017/images170927_1234_2017-09-27-10-58-17_44.bag
    /home/xesc/BAGFILES_BAG/emisari_27092017/images170927_1234_2017-09-27-10-58-49_45.bag"/>

  <arg name="camera" default="stereo_forward"/>
  <arg name="image" default="image_raw"/>

  <param name="use_sim_time" value="true"/>

  <node pkg="rosbag" type="play" name="player" args="--clock -d 5 -l $(arg bagfiles)"/>

  <!--group ns="$(arg camera)">
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc"/>
  </group-->

  <!--include file="$(find turbot)/launch/modules/stereo_pipeline.launch">
    <arg name="stereo" value="$(arg camera)" />
  </include-->

  <include file="$(find turbot)/launch/modules/stereo_pipeline.launch">
    <arg name="stereo" value="/vig/$(arg camera)" />
  </include>

  <!--group ns="/vig/$(arg camera)">
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc"/>
  </group-->

  <node pkg="image_preprocessing" type="vignetting_remover" name="vignetting_remover_l" output="screen">
    <param name="mean_filename" value="$(find image_preprocessing)/data/vignetting_mean_left.png"/>
    <param name="std_filename" value="$(find image_preprocessing)/data/vignetting_std_left.png"/>
    <param name="output_namespace" value="/vig/$(arg camera)/left"/>
    <remap from="image" to="$(arg camera)/left/$(arg image)"/>
  </node>

  <!--node pkg="image_preprocessing" type="vignetting_remover" name="vignetting_remover_r" output="screen">
    <param name="mean_filename" value="$(find image_preprocessing)/data/vignetting_mean_right.png"/>
    <param name="std_filename" value="$(find image_preprocessing)/data/vignetting_std_right.png"/>
    <param name="output_namespace" value="/vig/$(arg camera)/right"/>
    <remap from="image" to="$(arg camera)/right/$(arg image)"/>
  </node-->
  
  <node pkg="image_preprocessing" type="color_correction_image" name="color_correction_image" output="screen">
    <remap from="/color_correction_image/input" to="/vig/$(arg camera)/scaled_x2/left/image_rect_color"/>
    <remap from="/color_correction_image/output" to="/vig/$(arg camera)/scaled_x2/left/image_rect_color_corrected"/>
  </node>

  <!--node pkg="image_preprocessing" type="stereo_clahs_node" name="clahs_node" respawn="true" output="screen" >
    <remap from="stereo" to="/vig/$(arg camera)/scaled_x2"/>
  </node-->

</launch>
